elliptic-curve-crate/src/weierstrass.rs

```diff
@@ -4,8 +4,30 @@ pub mod curve;
 pub mod point;
 pub mod public_key;
 
-pub use curve::{Curve, ScalarBytes};
+pub use curve::Curve;
 pub use point::{
     CompressedCurvePoint, CompressedPointSize, UncompressedCurvePoint, UncompressedPointSize,
 };
 pub use public_key::PublicKey;
+
+use crate::{consts::U1, ScalarBytes};
+use core::ops::Add;
+use generic_array::ArrayLength;
+use subtle::{ConditionallySelectable, CtOption};
+
+/// Fixed-base scalar multiplication
+pub trait FixedBaseScalarMul: Curve
+where
+    <Self::ScalarSize as Add>::Output: Add<U1>,
+    CompressedCurvePoint<Self>: From<Self::Point>,
+    UncompressedCurvePoint<Self>: From<Self::Point>,
+    CompressedPointSize<Self::ScalarSize>: ArrayLength<u8>,
+    UncompressedPointSize<Self::ScalarSize>: ArrayLength<u8>,
```

Comment:
Oh `generic-array` trait bounds 😢

Reply:
By introducing an `AffineCoordinates` trait I was able to eliminate the `CompressedCurvePoint<Self>: From<Self::Point>` and `UncompressedCurvePoint<Self>: From<Self::Point>` bounds 🎉 

See #50